olive/systems/python_environment/python_environment_system.py

```diff
@@ -63,6 +64,8 @@ def __init__(
             self.environ["PATH"] = os.pathsep.join(self.config.prepend_to_path) + os.pathsep + self.environ["PATH"]
         if self.config.python_environment_path:
             self.environ["PATH"] = str(self.config.python_environment_path) + os.pathsep + self.environ["PATH"]
+        if self.config.olive_managed_env and platform.system() == "Linux":
+            self.environ["TMPDIR"] = self.environ["HOME"] + "TMP"
```

Comment:
two issues here
1. use os.path.join instead of + to generate the temp path
2. need specify the temp path in pip install